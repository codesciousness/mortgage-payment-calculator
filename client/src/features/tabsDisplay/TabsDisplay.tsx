import React, { useState } from 'react';
import Tabs from '@mui/material/Tabs';
import Tab from '@mui/material/Tab';
import Box from '@mui/material/Box';
import TabPanel from '../../components/tabPanel/TabPanel';
import MonthlyPayment from '../monthlyPayment/MonthlyPayment';
import AmortizationSchedule from '../amortizationSchedule/AmortizationSchedule';
import './TabsDisplay.css';

const TabsDisplay = (): JSX.Element => {
    const [value, setValue] = useState(0);

    const handleChange = (event: React.SyntheticEvent, newValue: number) => {
        setValue(newValue);
    };
    
    return (
        <Box sx={{ width: '100%' }}>
            <Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
                <Tabs 
                    value={value}
                    onChange={handleChange}
                    aria-label='tabs-display'
                    sx={{
                        '& .MuiButtonBase-root.MuiTab-root': {
                            fontWeight: 'bold',
                            textTransform: 'none'
                        },
                        '& .Mui-selected': {
                            color: 'DeepSkyBlue'
                        },
                        '& .MuiTabs-indicator': {
                            backgroundColor: 'DeepSkyBlue',
                            height: 3
                        }
                    }}
                >
                    <Tab label='Monthly payment'/>
                    <Tab label='Amortization'/>
                </Tabs>
            </Box>
            <TabPanel value={value} index={0}>
                <MonthlyPayment/>
            </TabPanel>
            <TabPanel value={value} index={1}>
                <AmortizationSchedule/>
            </TabPanel>
        </Box>
    );
};

export default TabsDisplay;