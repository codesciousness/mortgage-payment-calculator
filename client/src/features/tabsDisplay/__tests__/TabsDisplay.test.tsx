import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { create } from 'react-test-renderer';
import TabsDisplay from '../TabsDisplay';

jest.mock('../../monthlyPayment/MonthlyPayment', () => () => (<section data-testid="MonthlyPayment">Monthly Payment</section>));
jest.mock('../../amortizationSchedule/AmortizationSchedule', () => () => (<section data-testid="AmortizationSchedule">Amortization Schedule</section>));

describe('<TabsDisplay />', () => {
    test('TabsDisplay component is defined', () => {
        expect(<TabsDisplay/>).toBeDefined();
    });
    test('component matches the snapshot', () => {
        const component = create(<TabsDisplay/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders the MonthlyPayment component as the default tab', () => {
        render(<TabsDisplay/>);
        expect(screen.getByTestId('MonthlyPayment')).toBeInTheDocument();
        expect(screen.getByText('Monthly Payment')).toBeInTheDocument();
        expect(screen.queryByTestId('AmortizationSchedule')).not.toBeInTheDocument();
        expect(screen.queryByText('Amortization Schedule')).not.toBeInTheDocument();
    });
    test('renders the AmortizationSchedule component when the Amortization tab is clicked', () => {
        render(<TabsDisplay/>);
        fireEvent.click(screen.getByText('Amortization'));
        expect(screen.getByTestId('AmortizationSchedule')).toBeInTheDocument();
        expect(screen.getByText('Amortization Schedule')).toBeInTheDocument();
        expect(screen.queryByTestId('MonthlyPayment')).not.toBeInTheDocument();
        expect(screen.queryByText('Monthly Payment')).not.toBeInTheDocument();
    });
    test('renders the MonthlyPayment component when the Monthly payment tab is clicked', () => {
        render(<TabsDisplay/>);
        fireEvent.click(screen.getByText('Amortization'));
        expect(screen.getByTestId('AmortizationSchedule')).toBeInTheDocument();
        expect(screen.getByText('Amortization Schedule')).toBeInTheDocument();
        expect(screen.queryByTestId('MonthlyPayment')).not.toBeInTheDocument();
        expect(screen.queryByText('Monthly Payment')).not.toBeInTheDocument();
        fireEvent.click(screen.getByText('Monthly payment'));
        expect(screen.getByTestId('MonthlyPayment')).toBeInTheDocument();
        expect(screen.getByText('Monthly Payment')).toBeInTheDocument();
    });
});