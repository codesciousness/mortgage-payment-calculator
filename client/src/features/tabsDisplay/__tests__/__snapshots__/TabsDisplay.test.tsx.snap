// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabsDisplay /> component matches the snapshot 1`] = `
<section
  className="TabsDisplay"
  id="TabsDisplay"
>
  <div
    className="MuiBox-root css-1gsv261"
  >
    <div
      className="MuiTabs-root css-1bc84cv-MuiTabs-root"
    >
      <div
        className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
        onScroll={[Function]}
        style={
          Object {
            "marginBottom": -0,
            "overflow": "hidden",
          }
        }
      >
        <div
          aria-label="tabs-display"
          aria-orientation={null}
          className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-selected={true}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={0}
            type="button"
          >
            Monthly payment
            <span
              className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
              style={Object {}}
            />
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            Amortization
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-labelledby="tab-0"
    hidden={false}
    id="tabpanel-0"
    role="tabpanel"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiBox-root css-1ggs94k-MuiPaper-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        <section
          data-testid="MonthlyPayment"
        >
          Monthly Payment
        </section>
      </p>
    </div>
  </div>
  <div
    aria-labelledby="tab-1"
    hidden={true}
    id="tabpanel-1"
    role="tabpanel"
  />
</section>
`;
